### PR TITLE
Add option to provide additional locales to kirby-calendar and add locale documentation

### LINKED ISSUE #2648: [Enhancement] Cookbook Calendar doesn't specify the use of locales properly
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement

In the cookbook [calendar page](https://cookbook.kirby.design/#/home/showcase/calendar) we need to improve the documentation on how to use locales. As it is right now, the cookbook doesn't properly specify how to set the locale on the calendar component. 

The improvement must include: 
1. An example on how to se locale using the `@Injector` property and provider
2. An example on how to override the injected locale with a custom
3. Remove the restriction on only having `DA` or `EN` locales injected 

## Have you considered any alternatives?

No
## Are there any additional context?

No
<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [-] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by mark-drastrup**
## Which issue does this PR close?

This PR closes #2648

## What is the new behavior?
Users can now add additional locales (like Spanish, French etc..). The cookbook has also been updated with documentation on locales.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html [hunk 1]
```diff
@@ -44,4 +44,46 @@ <h4>Properties:</h4>
...
+    It can also be provided (or overridden if needed) in the component that uses
+    <code>kirby-calendar</code> like this:
+  </p>
+
+  <cookbook-code-viewer [ts]="componentProviderExample"></cookbook-code-viewer>
+
+  <h4>Adding additional existing locales:</h4>
+
+  <p>
+    To provide additional locales than the already built-in (Danish & English), simply import them
```
Comment (kodeaben) [inline apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html]: i think it needs to be  `other than`

File: apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html [hunk 2]
```diff
@@ -44,4 +44,46 @@ <h4>Properties:</h4>
...
+    from <code>date-fns/locale</code> and pass it to <code>kirby-calendar</code> in an object.
+  </p>
+
+  <cookbook-code-viewer [ts]="additionalLocalesExampleTS"></cookbook-code-viewer>
+  <cookbook-code-viewer [html]="additionalLocalesExampleHTML"></cookbook-code-viewer>
+
+  <h4>Adding custom locales:</h4>
```
Comment (kodeaben) [inline apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html]: is  colon (`:`) commonly used in the cookbook?

File: apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.ts [hunk 3]
```diff
@@ -3,6 +3,14 @@ import exampleWithCardHtml from '../../examples/calendar-example/calendar-card-e
...
 import { ApiDescriptionEvent } from '~/app/shared/api-description/api-description-events/api-description-events.component';
 import { ApiDescriptionProperty } from '~/app/shared/api-description/api-description-properties/api-description-properties.component';
+import {
+  additionalLocalesExampleHTML,
+  additionalLocalesExampleTS,
+  componentProviderExample,
+  customLocaleExampleHTML,
+  customLocaleExampleTS,
+  moduleProviderExample,
+} from '~/app/examples/calendar-example/calendar-locales-examples';
```
Comment (kodeaben) [inline apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.ts]: this import should be relative

File: libs/designsystem/src/lib/components/calendar/calendar.component.ts [hunk 4]
```diff
@@ -78,6 +78,7 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
   @Input() disablePastDates = false;
   @Input() disableFutureDates = false;
   @Input() alwaysEnableToday = false;
+  @Input() locales: { [key: string]: Locale } = {}; // Extra list of already existing locales
```
Comment (kodeaben) [inline libs/designsystem/src/lib/components/calendar/calendar.component.ts]: Should this comment be deleted`

**Review by kodeaben — DISMISSED** [review]
I'm surprised that there were no need for writing new tests, since we are allowing more locales?

File: apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html (hunk 2 above)
Comment (mark-drastrup) [reply apps/cookbook/src/app/showcase/calendar-showcase/calendar-showcase.component.html]: I checked and it depends on which page on the documentation you're looking at 😅 I have removed them from the calendar documentation :)

**Comment by mark-drastrup** [comment]
> I'm surprised that there were no need for writing new tests, since we are allowing more locales?

@kodeaben there actually wasn't any tests for locales at all. I have added some now, but I'm a bit uncertain on how to add tests for the custom locales. It requires a lot of code to make a custom locale, and I think it's a bit overkill to do it just for 1 test. We could write a test, where we pass an existing locale (French, Spanish etc.) to `customLocales` and test if that is being applied. Of course that doesn't test the real scenario for custom locales, but it will check if the locale that is passed to `customLocale` is being applied. What do you think? :)

File: apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+export const moduleProviderExample = `import {registerLocaleData } from '@angular/common';
```
Comment (kodeaben) [inline apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts:1]: format: `import { registerLocaleData } from '@angular/common';`

File: apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    providers: [
+        { provide: LOCALE_ID, useValue: 'en-DK' },
+    ],
+    ...
+})
+export class AppModule {}
+`;
+
+export const componentProviderExample = `import { Component, LOCALE_ID } from '@angular/core';
+import { Locale } from '@kirbydesign/designsystem/components/calendar/calendar.component';
```
Comment (kodeaben) [inline apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts]: shouldn't the import be like: `import { Locale } from '@kirbydesign/designsystem';`

File: apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    ...
+})
+export class MyComponent {
+    additionalLocales = { es, ja };
+}
+`;
+
+export const additionalLocalesExampleHTML = `<kirby-calendar [locales]="additionalLocales"></kirby-calendar>`;
+
+export const customLocaleExampleTS = `import { Locale } from '@kirbydesign/designsystem/components/calendar/calendar.component';
```
Comment (kodeaben) [inline apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts]: same

**Review by kodeaben — DISMISSED** [review]

File: apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts (hunk 6 above)
Comment (mark-drastrup) [reply apps/cookbook/src/app/examples/calendar-example/calendar-locales-examples.ts]: `Locale` wasn't exported through the calendar barrel file, but I have added it now and changed the import in the examples :)

**Review by kodeaben — APPROVED** [review]

**Pull request merged**